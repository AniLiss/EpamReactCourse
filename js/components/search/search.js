import React from 'react';
import './search.scss';
import {SearchButton} from '../search-btn/serch-btn';

export const Search = () => {
    return (
        <div className="search">
            <h2 className="search__title">Find your movie</h2>
            <input className="search__input" />
            <div className="search__filters">
                <h2 className="search__filters-title">Search by</h2>
                <button className="search__filters-item search__filters-item--active">Title</button>
                <button className="search__filters-item">Genre</button>
            </div>
            <SearchButton />
        </div>
    )
};
