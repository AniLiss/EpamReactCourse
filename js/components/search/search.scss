.search {
  display: flex;
  flex-wrap: wrap;

  &__input {
    width: 100%;
  }

  &__filters {
    display: flex;
    flex-grow: 1;
  }

  &__filters-title {
    margin: 0;
  }
}

@mixin layout-container {
  max-width: 1400px;
  margin: 0 auto;
}
