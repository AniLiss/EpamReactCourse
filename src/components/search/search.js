import React, {Component} from 'react';
import './search.scss';
import {Button} from '../search-btn/serch-btn';

export class Search extends Component {
    state = {
        isButtonActive: false
    };

    handleClick = () => {
        this.setState(() => ({
            isButtonActive: !this.state.isButtonActive
        }));
    };

    render() {
        const {isButtonActive} = this.state;
        return (
            <div className="search">
                <h2 className="search__title">Find your movie</h2>
                <input className="search__input" />
                <div className="search__filters">
                    <h2 className="search__filters-title">Search by</h2>
                    <Button dataId="Title" text="Title" active={!isButtonActive} onClick={this.handleClick} />
                    <Button dataId="Genre" text="Genre" active={isButtonActive} onClick={this.handleClick} />
                </div>
                <Button text="Search" active/>
            </div>
        )
    }
};
