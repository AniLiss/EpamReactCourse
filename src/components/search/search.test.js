import React from 'react';
import {shallow} from 'enzyme';
import { Search } from './search';

const wrapper = shallow(<Search />);

describe('Search', () => {
    it('is expected to render Search correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
});
