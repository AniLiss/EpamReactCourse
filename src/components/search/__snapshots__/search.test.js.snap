// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search is expected to render Search correctly 1`] = `
<div
  className="search"
>
  <h2
    className="search__title"
  >
    Find your movie
  </h2>
  <input
    className="search__input"
  />
  <div
    className="search__filters"
  >
    <h2
      className="search__filters-title"
    >
      Search by
    </h2>
    <Button
      active={true}
      text="Title"
    />
    <Button
      text="Genre"
    />
  </div>
  <Button
    active={true}
    text="Search"
  />
</div>
`;
